src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -390,10 +390,17 @@ private bool ShutdownHostIfUnhealthy()
         /// <param name="instance">The <see cref="IHost"/> instance to remove</param>
         private async Task Orphan(IHost instance, ILogger logger, CancellationToken cancellationToken = default)
         {
-            var scriptHost = (ScriptHost)instance?.Services.GetService<ScriptHost>();
-            if (scriptHost != null)
+            try
+            {
+                var scriptHost = (ScriptHost)instance?.Services.GetService<ScriptHost>();
+                if (scriptHost != null)
+                {
+                    scriptHost.HostInitializing -= OnHostInitializing;
+                }
+            }
+            catch (ObjectDisposedException)
             {
-                scriptHost.HostInitializing -= OnHostInitializing;
+                // If the instance is already disposed, we cannot access its services.
```

Comment:
Leaving this subscription is concerning. The change improve things, but doesn't resolve the issue of possible leaks.

Reply:
Can we at least have an issue tracking this?

Reply:
The fact that the host is disposed means that we've already gone through this codepath once. So the subscription has already been de-registered. I realize it's not guaranteed, that all disposals happen during our call to Orphan, but in practice it seems like they are.

Reply:
Just wanted to make sure, as I believe there is a chance that wouldn't be the case and there are things we can do to make this a bit more robust.

Reply:
Oh, yeah, for sure. This change doesn't make things worse, at least :-) If that thing's disposed, there's no way for us to access the ScriptHost with the way things are currently set up. I've opened an issue to see what we can do there: #4092.